### PR TITLE
Add world blocking for missions

### PR DESCRIPTION
**Pull request by MertUnverdi**

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java [hunk 1]
```diff
@@ -135,8 +135,6 @@
     public String calculatingIslands = "%prefix% &7Calculating %amount% Players islands, Estimated time: %minutes% Minutes and %seconds% Seconds";
     public String calculatingFinished = "%prefix% &7Calculating Players islands Finished";
     public String calculationAlreadyInProcess = "%prefix% &7Calculating Players islands already in Process";
-    public String netherIslandsDisabled = "%prefix% &7Nether islands have been disabled";
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:138]: we don't need them if nether islands is disabled it should send player to nether world

File: src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java [hunk 2]
```diff
@@ -30,13 +30,11 @@ public void onBucketEvent(PlayerBucketEvent event) {
...
         Optional<Island> island = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getBlockClicked().getLocation());
 
-        if (!island.isPresent()) return;
-        if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, IridiumSkyblock.getInstance().getPermissions().bucket, "bucket")) {
-            return;
-        }
-
-        event.setCancelled(true);
-        player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotUseBuckets.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+        island.ifPresent(value -> {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BucketListener.java:33]: here just tried collapse the code a bit

File: src/main/java/com/iridium/iridiumskyblock/listeners/EntityDamageListener.java [hunk 3]
```diff
@@ -31,12 +31,10 @@ public void onEntityDamageByEntity(EntityDamageByEntityEvent event) {
             Projectile projectile = (Projectile) event.getDamager();
             if (projectile.getShooter() instanceof Player) {
                 Player player = (Player) projectile.getShooter();
-                if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(player), IridiumSkyblock.getInstance().getPermissions().killMobs, "killMobs")) {
-                    return;
+                if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(player), IridiumSkyblock.getInstance().getPermissions().killMobs, "killMobs")) {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/EntityDamageListener.java:34]: and here too

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java [hunk 4]
```diff
@@ -135,8 +135,6 @@
...
-    public String endIslandsDisabled = "%prefix% &7End islands have been disabled";
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:139]: Why did you remove this?

File: src/main/java/com/iridium/iridiumskyblock/listeners/EntityPickupItemListener.java [hunk 5]
```diff
@@ -15,11 +17,11 @@
     public void onEntityPickupItem(PlayerPickupItemEvent event) {
         Player player = event.getPlayer();
         Optional<Island> island = IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getItem().getLocation());
-        if (!island.isPresent()) return;
 
-        if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), IridiumSkyblock.getInstance().getUserManager().getUser(player), IridiumSkyblock.getInstance().getPermissions().pickupItems, "pickupItems")) {
-            event.setCancelled(true);
-        }
+        island.ifPresent(value -> {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/EntityPickupItemListener.java:21]: and here..

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java [hunk 6]
```diff
@@ -53,16 +53,11 @@ public void onPlayerInteractEntity(PlayerInteractEntityEvent event) {
...
-            return;
-        }
-
-        if (IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, IridiumSkyblock.getInstance().getPermissions().interactEntities, "interactEntities")) {
-            return;
-        }
-
-        event.setCancelled(true);
-        player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotInteractEntities.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+        island.ifPresent(value -> {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java:56]: why we not using ! instead return thing

File: src/main/java/com/iridium/iridiumskyblock/configs/Missions.java [hunk 7]
```diff
@@ -18,6 +18,8 @@
  */
 public class Missions {
 
+    public List<String> blockedWorlds = Collections.emptyList();
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:21]: We already have three IridiumSkyblock worlds, what's the point in this additional option?

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerJoinListener.java [hunk 8]
```diff
@@ -29,7 +29,7 @@ public void onJoin(PlayerJoinEvent event) {
         if (player.isOp()) {
             Bukkit.getScheduler().runTaskLater(IridiumSkyblock.getInstance(), () ->
                             player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getConfiguration().prefix + " &7Thanks for using IridiumSkyblock, if you like the plugin, consider donating at &bwww.patreon.com/Peaches_MLG"))
-                    , 5);
+                    , 10L);
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerJoinListener.java:32]: i just wanted adding L here but i think we should increase it to 2 seconds may it too early before player join and connection finished

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java [hunk 9]
```diff
@@ -2,42 +2,42 @@
...
 
 import java.util.Objects;
+import java.util.Optional;
 
 public class PlayerPortalListener implements Listener {
 
     @EventHandler
     public void onPlayerPortal(PlayerPortalEvent event) {
         IslandManager islandManager = IridiumSkyblock.getInstance().getIslandManager();
-        IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getFrom()).ifPresent(island -> {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: we need this, shouldn't send player while netherIslands enabled (it does on 2.6.7)

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java [hunk 10]
```diff
@@ -10,10 +10,11 @@
 
     @EventHandler
     public void onPlayerTeleport(PlayerTeleportEvent event) {
-        if (event.getTo() == null) return;
-        IridiumSkyblock.getInstance().getIslandManager().getIslandViaLocation(event.getTo()).ifPresent(island ->
-                PlayerUtils.sendBorder(event.getPlayer(), island)
-        );
+        if (event.getTo() != null) {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java:13]: use ! instead return

File: src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java [hunk 11]
```diff
@@ -858,4 +858,11 @@ public World getEndWorld() {
         return Bukkit.getWorld(IridiumSkyblock.getInstance().getConfiguration().worldName + "_the_end");
     }
 
+    public boolean isIslandWorld(String world) {
```
Comment (MertUnverdi) [inline src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java:861]: i was needed this while making an addon to iridium, i was asked why we haven't it, now i make it but tooo late :(

File: src/main/java/com/iridium/iridiumskyblock/configs/Missions.java (hunk 7 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:21]: like kill mission, when player kill an entity on spawn or anywhere it increments the mission
someone wanted blocking it

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java (hunk 6 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java:56]: Is that a question?

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerJoinListener.java (hunk 8 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerJoinListener.java:32]: It's not

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java (hunk 10 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java:13]: Why not return?

File: src/main/java/com/iridium/iridiumskyblock/configs/Missions.java (hunk 7 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:21]: Yes, but you should rather check for one of the three world names that already exist instead of introducing a new one

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java (hunk 1 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:138]: @Peaches_MLG do we really want it to behave that way?

File: src/main/java/com/iridium/iridiumskyblock/configs/Missions.java (hunk 7 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:21]: I thought it too, but may users wants not blocking some worlds

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java (hunk 6 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java:56]: yea i but why asked at here idk

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java (hunk 10 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerTeleportListener.java:13]: again my bad english, i want say "used"

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java (hunk 1 above)
Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:138]: no I dont see that being useful

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java (hunk 9 above)
Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: no, if they arnt on the island it should just do them as normal

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: I mean like our border issue
When player built portal on boundaries it was teleporting to normal nether, this will ignore that

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: its will ignore sending player to normal nether/end when any issue

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java (hunk 1 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:138]: so, what should i do, didn't understand

Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:141]: Just block the portals and send this message

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java (hunk 9 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: I don't get your point, we don't have boundary issues anymore

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: mabey, will have any new issues?

File: src/main/java/com/iridium/iridiumskyblock/configs/Missions.java (hunk 7 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:23]: Why would we want to make it a lot harder and confusing for most people if maybe one person wants that feature?

Just imagine you're renaming the worlds in the configuration.yml and now you need to know that the names in the missions.yml also have to be updated...

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java (hunk 9 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: You're confusing me... I get your point but I don't know why we should expect these problems to re-appear

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java (hunk 1 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:141]: So if netherIslands is disabled portals wont work on islands?? who wants it?
im still thinking we shouldn't block it <3

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java (hunk 9 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: isn't possilbe? im just want blocking when couldn't find the island, may players can except the out of island and build portal on there?

File: src/main/java/com/iridium/iridiumskyblock/configs/Missions.java (hunk 7 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:23]: umm someone wants it i didn't think this side, you're right about this i'll pull back :panda:

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:23]: or if you have another thing?

Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:23]: Another thing? I probably don't

Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/configs/Missions.java:23]: Just use the three configured worlds from the configuration.yml

File: src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java (hunk 9 above)
Comment (dlsf) [reply src/main/java/com/iridium/iridiumskyblock/listeners/PlayerPortalListener.java:20]: Also not possible. I'd just revert that too
